Introduced auto_decode flag

cnavacch opened the pull request:
`auto_decode` flag for applying auto scaling and masking for NetCDF single or timestack files.

cnavacch committed:
changed name of auto_scale to auto_decode; introduced auto_decode in timestack for NetCDF files

bbauerma commented:
syntax looks good!
one suggestion: could we add an attribute or something that the auto_decode was successful? because if a user sets it True, and it doesn't work for some reason, it would be bad if this remains unnoticed. Or raise an error?

sebhahn commented:
Please create some tests before merging!

Commit:
introduced checks for decoding attributes; updated tests; added test checking decoding of an netCDF file and stack (#2)

Commit:
added auto-decoding functionality for GeoTiff files and stacks and some basic tests (#2)

cnavacch commented:
I have now re-worked the tests from the old `pyraster` package and implemented an automatic decoding procedure for NetCDF and GeoTIFF images and stacks. Thus, I see this merge request as valid and will complete it to create a first release of `veranda`. This release will be then used by `yeoda`, which will be also updated in compliance with the branch `scatsar_swi`.
